### DISCUSSION TITLE
Ring Intercom time to open

### DISCUSSION THREAD

**Discussion by developerGM** [tsightler/ring-mqtt#685] [Feature Requests]
It would be possible to reduce the time of pulse when click on "Unlock door"?
Now, when I click, the signal time is 3000ms

**Comment by tsightler** [comment]
As noted, please do not open issues for feature requests.  Feature requests should be discussed in the feature requests discussion topic first.  This issue will converted to a discussion.

**Comment by tsightler** [comment]
> It would be possible to reduce the time of pulse when click on "Unlock door"?
> Now, when I click, the signal time is 3000ms

I'm afraid I don't understand the request.  Do you mean the actual signal that is sent to the lock?  Is this something that is possible to change in the Ring app?  Or are you referring only to the state?  Also, why is this a problem?

**Reply by developerGM** [reply]
Yes, sorry, this is not an issues, absolutly

When I click on "open" from ha or from Ring app, the inpulse time is 3s
so I listen the sound of my door opening for 3secs

I found, checking on log of my Ring that there is a setting about this 3000ms but I don't know if this is writable or not

**Reply by tsightler** [reply]
> When I click on "open" from ha or from Ring app, the inpulse time is 3s

OK, so if it works the same in the Ring app then there's unlikely to be anything that can be done differently in ring-mqtt.  The API call in ring-client-api does not appear to have any options for setting to time, nor would I expect it to if that setting isn't exposed for the unlock call in the app.  This is largely answered in the FAQ, if you can't do it in the app, then it probably isn't possible in ring-mqtt as it's the same API.

It looks like there is already a feature requests for this on the Ring community forums so I would suggest continuing to add your voice there or by contacting Ring support.

**Reply by developerGM** [reply]
Ok thanks for your support

**Comment by developerGM** [comment]
To complete my issue, these are the attrs that I can read

`{'id': xxx, 'kind': 'intercom_handset_audio', 'description': 'Ingresso', 'location_id': 'xxx', 'schema_id': None, 'is_sidewalk_gateway': False, 'created_at': '2023-04-26T16:39:56Z', 'deactivated_at': None, 'owner': {'id': xxx, 'first_name': '', 'last_name': '', 'email': ''}, 'device_id': 'xxxx', 'time_zone': 'Europe/Rome', 'firmware_version': 'Up to Date', 'owned': True, 'ring_net_id': None, 'settings': {'features_confirmed': 3, 'show_recordings': True, 'recording_ttl': 180, 'recording_enabled': False, 'keep_alive': None, 'keep_alive_auto': 45.0, 'doorbell_volume': 8, 'enable_chime': 1, 'theft_alarm_enable': 0, 'use_cached_domain': 1, 'use_server_ip': 0, 'server_domain': 'fw.ring.com', 'server_ip': None, 'enable_log': 1, 'forced_keep_alive': None, 'mic_volume': 11, 'chime_settings': {'enable': True, 'type': 2, 'duration': 10}, 'intercom_settings': {'ring_to_open': False, 'predecessor': '{"make":"Cyfral","model":"SMART-5PG","wires":5}', 'config': '{"intercom_type": 0, "number_of_wires": 5, "autounlock_enabled": false, "analog": {"unlock_duration": 3000, "common_audio": false, "audio_term_io": 3, "audio_term_out": 3, "control_data_unlock1": "0x0880", "control_type_unlock1": 1, "control_data_unlock2": "0x0000", "control_type_unlock2": 0, "control_data_talk": "0x0000", "control_type_talk": 0, "control_data_listen": "0x0000", "control_type_listen": 0, "min_edge_count": 0, "ring_detect": 0}}', 'intercom_type': 'AF', 'replication': 0, 'unlock_mode': 0}, 'voice_volume': 11}, 'alerts': {'connection': 'online', 'ota_status': 'timeout'}, 'function': {'name': None}, 'subscriptions': {'event_types': []}, 'subscribed': False, 'battery_life': '66', 'features': {'cfes_eligible': False, 'motion_zone_recommendation': False, 'motions_enabled': True, 'show_recordings': True, 'show_vod_settings': True, 'rich_notifications_eligible': False, 'show_offline_motion_events': False, 'sheila_camera_eligible': None, 'sheila_camera_processing_eligible': None, 'dynamic_network_switching_eligible': False, 'chime_auto_detect_capable': False, 'show_24x7_lite': False}, 'metadata': {'legacy_fw_migrated': True, 'imported_from_amazon': False, 'is_sidewalk_gateway': False}}`

you can see `{"unlock_duration": 3000`

**Reply by tsightler** [reply]
Yes, but this is outside of the scope of ring-mqtt.  Based on what it looks like above it is a part of config property for intercom_settings.  My guess is that these are configured with pre-defined values based on selecting the intercom type during installation, but I could certainly be wrong about that.

It might be possible to set this value using standard Ring API calls (just because it's not implemented in the app doesn't mean the API doesn't support it, just means it's more difficult to figure out how to set it), but I don't think ring-client-api currently has a generic setDeviceSettings() function for intercoms like what is available for cameras, although my guess is the back-end API would be the same.  However, as I don't own an intercom, it's not something I would be able to poke at myself.  If you decide to dig into it and figure out how to change this value, you could submit a PR to ring-client-api and eventually that capability could make its way into ring-mqtt.
